specs/_features/eip7732/beacon-chain.md

```diff
@@ -457,8 +433,102 @@ def is_parent_block_full(state: BeaconState) -> bool:
     return state.latest_execution_payload_header.block_hash == state.latest_block_hash
 ```
 
+### Misc
+
+#### New `remove_flag`
+
+```python
+def remove_flag(flags: ParticipationFlags, flag_index: int) -> ParticipationFlags:
+    flag = ParticipationFlags(2**flag_index)
+    return flags & ~flag
+```
+
+#### New `compute_balance_weighted_selection`
+
+```python
+def compute_balance_weighted_selection(
+    state: BeaconState,
+    indices: Sequence[ValidatorIndex],
+    seed: Bytes32,
+    size: uint64,
+    shuffle_indices: bool = True,
+) -> Sequence[ValidatorIndex]:
+    """
+    Return `size` indices sampled by effective balance, using `indices`
+    as candidates. If `shuffle_indices` is `True`, candidate indices
+    are themselves sampled from `indices` by shuffling it, otherwise
+    `indices` is traversed in order.
+    """
+    total = uint64(len(indices))
+    assert total > 0
+    selected: List[ValidatorIndex] = []
+    i = uint64(0)
+    while len(selected) < size:
+        next_index = i % total
+        if shuffle_indices:
+            next_index = compute_shuffled_index(next_index, total, seed)
+        candidate_index = indices[next_index]
+        if compute_balance_weighted_acceptance(state, candidate_index, seed, i):
+            selected.append(candidate_index)
+        i += 1
+    return selected
```

Comment:
This can contain duplicates. Is this acceptable? I'm not sure that it is.

Reply:
To answer my own question, I suppose the answer is yes. The current implementations of `compute_proposer_indices` & `get_next_sync_committee_indices` allow this.

Reply:
It explicitly states it here too:
https://github.com/ethereum/consensus-specs/blob/35f8191a923706e5f702267b555be65593613d59/specs/electra/beacon-chain.md?plain=1#L649